src-ui.v3/src/Main.hs

```diff
@@ -777,11 +736,72 @@ pkgTagList m = Map.fromListWith (List.++) $ do
 joinE :: forall t m a. (Reflex t, MonadHold t m) => Event t (Event t a) -> m (Event t a)
 joinE = fmap switch . hold never
 
-button_ :: forall t m. (DomBuilder t m, PostBuild t m) => T.Text -> m (Event t ())
-button_ t = do
+clickElement_ :: forall t m. (DomBuilder t m, PostBuild t m) => T.Text -> T.Text -> m (Event t ())
+clickElement_ elm t = do
   let cfg = (def :: ElementConfig EventResult t (DomBuilderSpace m))
         & elementConfig_eventSpec %~ addEventSpecFlags (Proxy :: Proxy (DomBuilderSpace m)) Click (\_ -> preventDefault)
-  (e, _) <- element "button" cfg $ text t
+  (e, _) <- element elm cfg $ text t
   pure $ domEvent Click e
 
+stripSearch :: JSS.JSString -> JSS.JSString
+stripSearch sJ
+  | Just sJ'  <- JSS.stripPrefix "^" sJ = sJ'
+  | Just sJ' <- JSS.stripSuffix "$" sJ  = sJ'
+  | otherwise                           = sJ
+
+filterBySearchBox :: [JSS.JSString] -> JSS.JSString -> [JSS.JSString]
+filterBySearchBox pkgs sJ
+  | JSS.isInfixOf "^" sJ = filter (JSS.isPrefixOf (stripSearch sJ)) pkgs
```

Comment:
checking `isInfixOf` and then stripping the prefix or suffix via `stripPrefix`/`stripSuffix` has two problems:

1. it mixes different semantics; isInfixOf will match in more cases than `stripPrefixOf`/`stripSuffixOf` will match on
2. it wastes on the benefit of using `stripPrefixOf`/`stripSuffixOf` in the first place: doing the check and the stripping in the very same step in an optimal way

Reply:
@hvr I think this file in this commit is already outdated. I have pushed my latest commit and have change `filterBySearchBox` entirely